packages/plugins/src/worker/push-notification.ts

```diff
@@ -0,0 +1,99 @@
+import { WorkerDirector, WorkerAdapter } from '@unchainedshop/core-worker';
+import { createLogger } from '@unchainedshop/logger';
+import { IWorkerAdapter } from '@unchainedshop/types/worker.js';
+import webPush from 'web-push';
+
+const { PUSH_NOTIFICATION_PUBLIC_KEY, PUSH_NOTIFICATION_PRIVATE_KEY } = process.env;
+
+const logger = createLogger('unchained:plugins:worker:push-notification');
+
+const PushNotificationWorkerPlugin: IWorkerAdapter<
+  {
+    subscription: any;
+    title: string;
+    body: string;
+    url: string;
+  },
+  any
+> = {
+  ...WorkerAdapter,
+
+  key: 'shop.unchained.worker-plugin.push-notification',
+  label: 'Push Notification',
+
+  version: '1.0',
+
+  type: 'PUSH',
+
+  doWork: async ({ subscription, title, body, url }) => {
+    logger.debug(`${PushNotificationWorkerPlugin.key} -> doWork: ${title} -> ${body}`);
+    if (!PUSH_NOTIFICATION_PUBLIC_KEY)
+      return {
+        success: false,
+        error: {
+          name: 'VAPID_PUBLIC_KEY_REQUIRED',
+          message: 'vapidPublicKey key required to send push notifications',
+        },
+      };
+    if (!PUSH_NOTIFICATION_PRIVATE_KEY)
+      return {
+        success: false,
+        error: {
+          name: 'VAPID_PRIVATE_KEY_REQUIRED',
+          message: 'vapidPrivateKey key are required to send push notifications',
+        },
+      };
+
+    if (!subscription) {
+      return {
+        success: false,
+        error: {
+          name: 'USER_SUBSCRIPTION_OBJECT_REQUIRED',
+          message: 'PUSH service subscription required',
+        },
+      };
+    }
+
+    const options = {
+      vapidDetails: {
+        subject: url,
+        publicKey: PUSH_NOTIFICATION_PUBLIC_KEY,
+        privateKey: PUSH_NOTIFICATION_PRIVATE_KEY,
+      },
+      TTL: 60,
+    };
+
+    try {
+      await webPush.sendNotification(
+        subscription,
+        JSON.stringify({
```

Comment:
There is 2 options that I think we should forward to webPush.sendNotification if specified in input of work item:

urgency is to indicate to the push service whether to send the notification immediately or prioritize the recipient’s device power considerations for delivery. Provide one of the following values: very-low, low, normal, or high. To attempt to deliver the notification immediately, specify high.

topic optionally provide an identifier that the push service uses to coalesce notifications. Use a maximum of 32 characters from the URL or filename-safe Base64 characters sets.

+ I'd not stringify { title, body, url } and instead just accept a param "payload" in input and forward it to JSON.stringify(payload). That way the worker has full flexibility on how to handle the data.